torch/nn/parallel/distributed.py

```diff
@@ -625,24 +637,26 @@ class DistributedDataParallel(Module, Joinable):
     # used to track whether the given thread is inside ddp forward for torchdynamo purposes
     _active_ddp_module: Optional["DistributedDataParallel"] = None
 
+    reducer: dist.Reducer
+
     def __init__(
         self,
-        module,
-        device_ids=None,
-        output_device=None,
-        dim=0,
-        broadcast_buffers=True,
-        process_group=None,
-        bucket_cap_mb=None,
-        find_unused_parameters=False,
-        check_reduction=False,
-        gradient_as_bucket_view=False,
-        static_graph=False,
+        module: Module,
+        device_ids: Optional[Sequence[Union[int, device]]] = None,
+        output_device: Optional[Union[int, device]] = None,
+        dim: int = 0,
+        broadcast_buffers: bool = True,
+        process_group: "Optional[ProcessGroup]" = None,
+        bucket_cap_mb: float = 25,
+        find_unused_parameters: bool = False,
+        check_reduction: bool = False,
+        gradient_as_bucket_view: bool = False,
+        static_graph: bool = False,
         delay_all_reduce_named_params=None,
         param_to_hook_all_reduce=None,
         mixed_precision: Optional[_MixedPrecision] = None,
         device_mesh=None,
```

Comment:
`device_mesh` didn't exist when `distributed.pyi` was removed, which is why I didn't add it. But I could add a type for it if you like.

Reply:
Added it.